packages/neptune-extension/src/common/state/notebook/reducer.ts

```diff
@@ -1,24 +1,84 @@
-import { Reducer } from 'redux';
 import { NotebookActions } from './actions'
 import { NotebookDTO } from 'generated/leaderboard-client/src/models';
 
 interface NotebookState {
-  loaded: boolean
+  fetchStatus: 'none' | 'pending' | 'success' | 'failure'
```

Comment:
I think we should create:
type `FetchStatus = 'none' | 'pending' | 'success' || 'failure'` and use it in all cases.
Especially when you check against this status in components.